internal/rego/rego.go

```diff
@@ -2,196 +2,283 @@ package rego
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 	"path/filepath"
-	"regexp"
 	"sort"
 	"strings"
 
-	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/loader"
 )
 
-// File is a parsed Rego file.
-type File struct {
-	FilePath       string
-	PackageName    string
-	ImportPackages []string
-	Contents       string
-	RuleNames      []string
-	Comments       []string
+// Severity describes the severity level of the rego file.
+type Severity string
+
+// The defined severity levels represent the valid severity
+// levels that a rego file can have.
+const (
+	Violation Severity = "Violation"
+	Warning   Severity = "Warning"
```

Comment:
I went back and forth on this. Using it within the package itself, I tend to agree with there not beginning much value. The intent was more for consumers to be able to compare types rather than magic strings.

e.g. .Severity() == "Warning" vs .Severity() == rego.Warning

Reply:
Sorry, I made an incorrect assumption about Go's handling of that statement.